packages/core/src/actions/transactions/sendTransaction.ts

```diff
@@ -1,54 +1,111 @@
 import { providers } from 'ethers'
 
-import { getClient } from '../../client'
 import {
   ChainMismatchError,
   ConnectorNotFoundError,
   ProviderRpcError,
   UserRejectedRequestError,
 } from '../../errors'
-import { Chain } from '../../types'
+import { fetchSigner, getNetwork } from '../accounts'
 
-export type SendTransactionArgs = {
+export type SendTransactionPreparedRequest = {
   /**
-   * Chain id to use for write
-   * If signer is not active on this chain, it will attempt to programmatically switch
-   */
-  chainId?: number
-  /** Object to use when creating transaction */
+   * `dangerouslyUnprepared`: Allow to pass through an unprepared `request`. Note: This has
+   * [UX pitfalls](/docs/prepare-hooks/intro#ux-pitfalls-without-prepare-hooks), it is highly recommended
+   * to not use this and instead prepare the request upfront using the `prepareSendTransaction` function.
+   *
+   * `prepared`: The request has been prepared with parameters required for sending a transaction
+   * via the `prepareSendTransaction` function
+   * */
+  mode: 'prepared'
+  /** The prepared request for sending a transaction. */
+  request: providers.TransactionRequest & {
+    to: NonNullable<providers.TransactionRequest['to']>
+    gasLimit: NonNullable<providers.TransactionRequest['gasLimit']>
+  }
+}
+export type SendTransactionUnpreparedRequest = {
+  mode: 'dangerouslyUnprepared'
+  /** The unprepared request for sending a transaction. */
   request: providers.TransactionRequest
 }
 
-export type SendTransactionResult = providers.TransactionResponse
+export type SendTransactionArgs = {
+  /** Chain ID used to validate if the signer is connected to the target chain */
+  chainId?: number
+} & (SendTransactionPreparedRequest | SendTransactionUnpreparedRequest)
+
+export type SendTransactionResult = {
+  hash: providers.TransactionResponse['hash']
+  wait: providers.TransactionResponse['wait']
+}
```

Comment:
The original `signer.sendTransaction` made an additional request to the RPC to fetch the transaction data after the transaction was sent. Since this RPC request can be slow & not all consumers need the whole object, I thought it would be better to only return what is given from the unchecked `signer.sendTransaction`, and then if the consumer needs more properties from the transaction, then they can call `useWaitForTransaction` with `confirmations: 0`. [See here](https://github.com/tmm/wagmi/pull/658/files/dc3c388d31f9ceb830f77f9cd7e5e274dc96ed2e#diff-93ee9dd40c1bffb44416caccad93c8bb95d55c09df0156aa2a41855c140be6c7R1)